runtime/doc/lsp.txt

```diff
@@ -1321,6 +1321,58 @@ save({lenses}, {bufnr}, {client_id})                 *vim.lsp.codelens.save()*
       • {client_id}  (number)
 
 
+==============================================================================
+Lua module: vim.lsp.semantic_tokens                      *lsp-semantic_tokens*
+
+                                           *vim.lsp.semantic_tokens.on_attach()*
```

Comment:
These will be overwritten by the docstrings in your implementation.

Also, I don't think we should officially expose them. If you need to use them in other modules, that's ok, but we should still mark them as `---@private`. The idea is that nothing here is user-facing (although plugins may use the functions under a big, fat NO WARRANTY disclaimer).